typescript/sdk/src/token/schemas.ts

```diff
@@ -71,7 +95,72 @@ export const WarpRouteDeployConfigSchema = z
     const entries = Object.entries(configMap);
     return (
       entries.some(
-        ([_, config]) => isCollateralConfig(config) || isNativeConfig(config),
+        ([_, config]) =>
+          isCollateralConfig(config) ||
+          isCollateralRebaseConfig(config) ||
+          isNativeConfig(config),
       ) || entries.every(([_, config]) => isTokenMetadata(config))
     );
-  }, `Config must include Native or Collateral OR all synthetics must define token metadata`);
+  }, WarpRouteDeployConfigSchemaErrors.NO_SYNTHETIC_ONLY)
+  .transform((warpRouteDeployConfig, ctx) => {
+    const hasCollateralRebase = Object.entries(warpRouteDeployConfig).find(
+      ([_, config]) => isCollateralRebaseConfig(config),
+    );
+    if (!hasCollateralRebase) return warpRouteDeployConfig;
+
+    if (!isCollateralRebasePairedCorrectly(warpRouteDeployConfig)) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: WarpRouteDeployConfigSchemaErrors.ONLY_SYNTHETIC_REBASE,
+      });
+
+      // This is a special symbol you can use to
+      // return early from the transform function.
+      // It has type `never` so it does not affect the
+      // inferred return type.
+      return z.NEVER;
+    }
+
+    const modifiedConfigMap = resolveCollateralChainName(warpRouteDeployConfig);
+    return modifiedConfigMap;
+  });
+
+function isCollateralRebasePairedCorrectly(
+  warpRouteDeployConfig: Record<string, TokenRouterConfig>,
+): boolean {
+  // Filter out all the non-collateral rebase configs to check if they are only synthetic rebase tokens
+  const otherConfigs = Object.entries(warpRouteDeployConfig).filter(
+    ([_, config]) => !isCollateralRebaseConfig(config),
+  );
+
+  if (otherConfigs.length === 0) return false;
+
+  // The other configs MUST be synthetic rebase
+  const allOthersSynthetic = otherConfigs.every(([_, config], _index) =>
+    isSyntheticRebaseConfig(config),
+  );
+  return allOthersSynthetic;
+}
+
+/**
+ * Resolves the collateral domain name, and update each synthetic rebase tokens
+ *
+ * @param warpRouteDeployConfig - The `WarpRouteDeployConfig` object to be modified.
+ * @returns The modified WarpRouteDeployConfig with the `collateralChainName` property set for `syntheticRebase` tokens, or an unmodified WarpRouteDeployConfig
+ */
+function resolveCollateralChainName(
+  warpRouteDeployConfig: Record<string, TokenRouterConfig>,
+): Record<string, TokenRouterConfig> {
+  const collateralRebaseEntries = Object.entries(warpRouteDeployConfig).find(
+    ([_, config]) => config.type === TokenType.collateralVaultRebase,
+  );
+
+  if (!collateralRebaseEntries) return warpRouteDeployConfig;
```

Comment:
I would expect this to throw
when would we call this function otherwise?

Reply:
because other TokenTypes rely on the schema, we want to passthrough the config when it's not a collateralVaultRebase